<a href="https://colab.research.google.com/github/VENKATB3/Sumanth1_Data690/blob/main/Part-B/Part_B_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part B Final
* Name : Venkata Sumanth Teja Bitragunta
* Date : 04/09/2022
* Campus Id : KT99571

# Energy and Mining
Mining is first and foremost a source of mineral commodities that all countries find essential for maintaining and improving their standards of living. Mined materials are needed to construct roads and hospitals, to build automobiles and houses, to make computers and satellites, to generate electricity, and to provide the many other goods and services that consumers enjoy.

# Why is it important to you and/or to others?
* Electricity is an essential part of modern life and important to the country economy. People use electricity for lighting, heating, cooling, and refrigeration and for operating appliances, computers, electronics, machinery, and public transportation systems.
* I would like to give the breif description of energy and mining including how much eletricity is used, how much of poulation is using eletricity and total production of eletricity.

# Analysis
* Data Source : https://www.worlddev.xyz/
* Topics : Energy and Mining
* Countries : USA, Canada, Sri Lanka, Pakistan, Afghanistan
* Time plot between : 2011 to 2018 


In [ ]:
import pandas as pd
import plotly
import plotly.express as px
DATA_URL = "https://raw.githubusercontent.com/DATA-690-SPRING-2022/Sumanth1_Data690/main/Part-B/wdi_data.csv"
DATA_URL = "https://raw.githubusercontent.com/DATA-690-SPRING-2022/Sumanth1_Data690/main/Part-B/wdi_Urban%20data.csv"
DATA_URL = "https://raw.githubusercontent.com/DATA-690-SPRING-2022/Sumanth1_Data690/main/Part-B/wdi_Rural%20data.csv"

In [ ]:
df = pd.read_csv(DATA_URL)

df.sample(10)

Unnamed: 0  Year       value          indicator Country Code  \
1             1  2012   21.424611     EG.ELC.FOSL.ZS          CAN   
81           81  2016   55.810063  EG.ELC.ACCS.RU.ZS          PAK   
82           82  2017   54.207423  EG.ELC.ACCS.RU.ZS          PAK   
63           63  2014   86.500512  EG.ELC.ACCS.RU.ZS          AFG   
59           59  2018  100.000000     EG.ELC.ACCS.ZS          USA   
113         113  2016  100.000000  EG.ELC.ACCS.UR.ZS          CAN   
13           13  2014   60.797689     EG.ELC.FOSL.ZS          LKA   
56           56  2015  100.000000     EG.ELC.ACCS.ZS          USA   
66           66  2017   97.091973  EG.ELC.ACCS.RU.ZS          AFG   
75           75  2018  100.000000  EG.ELC.ACCS.RU.ZS          CAN   

      Country Name         Region         Income Group    Lending Type  
1           Canada  North America          High income  Not classified  
81        Pakistan     South Asia  Lower middle income           Blend  
82        Pakistan     South Asia  Lower middle income           Blend  
63     Afghanistan     South Asia           Low income             IDA  
59   United States  North America          High income  Not classified  
113         Canada  North America          High income  Not classified  
13       Sri Lanka     South Asia  Lower middle income            IBRD  
56   United States  North America          High income  Not classified  
66     Afghanistan     South Asia           Low income             IDA  
75          Canada  North America          High income  Not classified

In [ ]:
df_2016_electricity_all = df.query("Year == 2016").query("indicator == 'EG.ELC.ACCS.ZS'")
df_2016_electricity_all.sample(5)

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
33          33  2016  100.00000  EG.ELC.ACCS.ZS          CAN         Canada   
49          49  2016   97.50000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
57          57  2016  100.00000  EG.ELC.ACCS.ZS          USA  United States   
41          41  2016   71.58532  EG.ELC.ACCS.ZS          PAK       Pakistan   
25          25  2016   97.70000  EG.ELC.ACCS.ZS          AFG    Afghanistan   

           Region         Income Group    Lending Type  
33  North America          High income  Not classified  
49     South Asia  Lower middle income            IBRD  
57  North America          High income  Not classified  
41     South Asia  Lower middle income           Blend  
25     South Asia           Low income             IDA

In [ ]:
df_2016_electricity_all_sort = df_2016_electricity_all.sort_values(by="value", ascending=False)

df_2016_electricity_all_sort.head()

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
33          33  2016  100.00000  EG.ELC.ACCS.ZS          CAN         Canada   
57          57  2016  100.00000  EG.ELC.ACCS.ZS          USA  United States   
25          25  2016   97.70000  EG.ELC.ACCS.ZS          AFG    Afghanistan   
49          49  2016   97.50000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
41          41  2016   71.58532  EG.ELC.ACCS.ZS          PAK       Pakistan   

           Region         Income Group    Lending Type  
33  North America          High income  Not classified  
57  North America          High income  Not classified  
25     South Asia           Low income             IDA  
49     South Asia  Lower middle income            IBRD  
41     South Asia  Lower middle income           Blend

In [ ]:
fig = px.bar(
    df_2016_electricity_all_sort, 
    x="Country Name", 
    labels={"value":"2016 Bar Plot of all countries electricity percentage", "Country Name":""},
    y="value",
    color="Country Name",
    height=800,
    template=list(plotly.io.templates.keys())[5]
)

fig.update_layout(showlegend=False)

fig.show()

# Access to electricity, urban(% of Urban population)

In [ ]:
df_2016_Urban_all = df.query("indicator == 'EG.ELC.ACCS.UR.ZS'")
df_2016_Urban_all = df_2016_Urban_all[df_2016_Urban_all["Country Code"].isin(["USA","CAN", "LKA", "PAK","AFG"])]

fig = px.line(
    df_2016_Urban_all, 
    x="Year", 
    labels={"value":"electricity % of urban population"},
    y="value",
    color="Country Name",
    height=800,
    template=list(plotly.io.templates.keys())[5],
    title="TOP 5 Countries by 2016 Access to electricity (% of population)"

)


fig.update_layout(showlegend=True)

fig.show()

* I used a Line Graph to show the difference in urban population using eletricity between Canada, United States, Afghanistan, Pakistan and Sri Lanka over time.
* People in USA and Canada were using the 100% of the electricity. USA and Canada were utilizing the electricity in all aspects.
* There was very much fluctuation in Afghanistan and from the year 2011 to 2018 there are severe up and downs in the usage of energy.
* Urban population in Pakistan and Sri Lanka were slightly increasing in terms of eletrical usage.

# Access to electricity, rural(% of rural population)

In [ ]:
df_2016_Rural_all = df.query("indicator == 'EG.ELC.ACCS.RU.ZS'")
df_2016_Rural_all = df_2016_Rural_all[df_2016_Rural_all["Country Code"].isin(["USA","CAN", "LKA", "PAK","AFG"])]

fig = px.line(
    df_2016_Urban_all, 
    x="Year", 
    labels={"value":"electricity % of Rural population"},
    y="value",
    color="Country Name",
    height=800,
    template=list(plotly.io.templates.keys())[5],
    title="TOP 5 Countries by 2016 Access to electricity (% of population)"

)


fig.update_layout(showlegend=True)

fig.show()

* Rural people in USA and Canada were constantly using the electricity. Total Population in both of the countries were using the electricity.
* Aghanistan is constantly fluctuating in both rural and urban areas.
* Srilanka and Pakistan were constantly maintaining the same fluctuation from the year 2011 to 2018

# Access to electricity, urban(% of urban population)

In [ ]:
df_2016_Urban_all = df.query("Year == 2016").query("indicator == 'EG.ELC.ACCS.UR.ZS'")
df_2016_Urban_all.sample(5)

Unnamed: 0  Year       value          indicator Country Code  \
137         137  2016  100.000000  EG.ELC.ACCS.UR.ZS          USA   
113         113  2016  100.000000  EG.ELC.ACCS.UR.ZS          CAN   
121         121  2016   99.347229  EG.ELC.ACCS.UR.ZS          PAK   
105         105  2016   99.500000  EG.ELC.ACCS.UR.ZS          AFG   
129         129  2016   99.000000  EG.ELC.ACCS.UR.ZS          LKA   

      Country Name         Region         Income Group    Lending Type  
137  United States  North America          High income  Not classified  
113         Canada  North America          High income  Not classified  
121       Pakistan     South Asia  Lower middle income           Blend  
105    Afghanistan     South Asia           Low income             IDA  
129      Sri Lanka     South Asia  Lower middle income            IBRD

In [ ]:
df_2016_Urban_all_sort = df_2016_electricity_all.sort_values(by="value", ascending=False)

df_2016_Urban_all_sort.head()

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
33          33  2016  100.00000  EG.ELC.ACCS.ZS          CAN         Canada   
57          57  2016  100.00000  EG.ELC.ACCS.ZS          USA  United States   
25          25  2016   97.70000  EG.ELC.ACCS.ZS          AFG    Afghanistan   
49          49  2016   97.50000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
41          41  2016   71.58532  EG.ELC.ACCS.ZS          PAK       Pakistan   

           Region         Income Group    Lending Type  
33  North America          High income  Not classified  
57  North America          High income  Not classified  
25     South Asia           Low income             IDA  
49     South Asia  Lower middle income            IBRD  
41     South Asia  Lower middle income           Blend

In [ ]:
fig = px.pie(df_2016_Urban_all_sort, values='value', names='Country Name', title='TOP 5 Countries by 2016 Access to electricity, Urban (% of urban population).',template=list(plotly.io.templates.keys())[5])
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(showlegend=True)

fig.show()

USA and Canada has population who uses electricity 100% in every movement. Pakistan and Srilanka are the next in which more than 95% of urban population were using the electricity. Afghanistan is the only country with the Less percentage of urban population using electricity.

# Access to electricity, rural(% of rural population)

In [ ]:
df_2011_Rural_all = df.query("Year == 2011").query("indicator == 'EG.ELC.ACCS.RU.ZS'")
df_2011_Rural_all.sample(5)

Unnamed: 0  Year       value          indicator Country Code  \
68          68  2011  100.000000  EG.ELC.ACCS.RU.ZS          CAN   
92          92  2011  100.000000  EG.ELC.ACCS.RU.ZS          USA   
76          76  2011   56.425752  EG.ELC.ACCS.RU.ZS          PAK   
60          60  2011   29.572881  EG.ELC.ACCS.RU.ZS          AFG   
84          84  2011   85.740489  EG.ELC.ACCS.RU.ZS          LKA   

     Country Name         Region         Income Group    Lending Type  
68         Canada  North America          High income  Not classified  
92  United States  North America          High income  Not classified  
76       Pakistan     South Asia  Lower middle income           Blend  
60    Afghanistan     South Asia           Low income             IDA  
84      Sri Lanka     South Asia  Lower middle income            IBRD

In [ ]:
df_2011_Rural_all_sort = df_2016_electricity_all.sort_values(by="value", ascending=False)

df_2011_Rural_all_sort.head()

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
33          33  2016  100.00000  EG.ELC.ACCS.ZS          CAN         Canada   
57          57  2016  100.00000  EG.ELC.ACCS.ZS          USA  United States   
25          25  2016   97.70000  EG.ELC.ACCS.ZS          AFG    Afghanistan   
49          49  2016   97.50000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
41          41  2016   71.58532  EG.ELC.ACCS.ZS          PAK       Pakistan   

           Region         Income Group    Lending Type  
33  North America          High income  Not classified  
57  North America          High income  Not classified  
25     South Asia           Low income             IDA  
49     South Asia  Lower middle income            IBRD  
41     South Asia  Lower middle income           Blend

In [ ]:
fig = px.pie(df_2011_Rural_all_sort, values='value', names='Country Name', title='TOP 5 Countries by 2011 Access to electricity, Rural (% of rural population).',template=list(plotly.io.templates.keys())[5])
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(showlegend=True)

fig.show()

From the above pie chart we can analyse how much rural population were using the eletricity. As from we know USA and Canada were having 100% of rural population using electricity. 86% of Srilanka rural population were using the electricity. Pakistan has the poulation of 57% were using the electricity in that country. Moreover, Afghanistan is the only country with less percentage of rural population using electricity which is approximately 29%.

# Access to electricity(% of population)

In [ ]:
df_2011_electricity_all = df.query("Year == 2011").query("indicator == 'EG.ELC.ACCS.ZS'")
df_2011_electricity_all.sample(5)

Unnamed: 0  Year       value       indicator Country Code   Country Name  \
52          52  2011  100.000000  EG.ELC.ACCS.ZS          USA  United States   
20          20  2011   43.222019  EG.ELC.ACCS.ZS          AFG    Afghanistan   
44          44  2011   87.760000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
36          36  2011   70.880081  EG.ELC.ACCS.ZS          PAK       Pakistan   
28          28  2011  100.000000  EG.ELC.ACCS.ZS          CAN         Canada   

           Region         Income Group    Lending Type  
52  North America          High income  Not classified  
20     South Asia           Low income             IDA  
44     South Asia  Lower middle income            IBRD  
36     South Asia  Lower middle income           Blend  
28  North America          High income  Not classified

In [ ]:
df_2011_electricity_all_sort = df_2011_electricity_all.sort_values(by="value", ascending=False)

df_2011_electricity_all_sort.head()

Unnamed: 0  Year       value       indicator Country Code   Country Name  \
28          28  2011  100.000000  EG.ELC.ACCS.ZS          CAN         Canada   
52          52  2011  100.000000  EG.ELC.ACCS.ZS          USA  United States   
44          44  2011   87.760000  EG.ELC.ACCS.ZS          LKA      Sri Lanka   
36          36  2011   70.880081  EG.ELC.ACCS.ZS          PAK       Pakistan   
20          20  2011   43.222019  EG.ELC.ACCS.ZS          AFG    Afghanistan   

           Region         Income Group    Lending Type  
28  North America          High income  Not classified  
52  North America          High income  Not classified  
44     South Asia  Lower middle income            IBRD  
36     South Asia  Lower middle income           Blend  
20     South Asia           Low income             IDA

In [ ]:
fig = px.pie(df_2011_Rural_all_sort, values='value', names='Country Name', title='TOP 5 Countries by 2011 Access to electricity,(% of population).',template=list(plotly.io.templates.keys())[5])
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(showlegend=True)

fig.show()

From the above pie chart we can see that Both USA and Canada were having the rural and uraban 100% of poulation were using the electricity. When look at the total population in Srilanka only 87% of population using the electricity and with pakistan only 70% of the total population were using the electricity for their needs. Afghanistan is the only country with less then 50% of the total population using the electricity.

# Conclusion

Overall there is lot of usage in every country but when compared with the top countries with highest usage of electricity lower countries usage is verry less. Due to, there is no severe action by the government. All countries are trying very hard to overcome this situation but many of them are facing many problems like shortage of water and fossil fuels. Countries have to try more to overcome this situation by using the solar and wind energy it will be very much helpful so that, there can be no other emissions like some distinct gases.